nifi-commons/nifi-properties/src/main/java/org/apache/nifi/util/NiFiProperties.java

```diff
@@ -1707,6 +1709,17 @@ public boolean isZooKeeperClientSecure() {
         return Boolean.parseBoolean(clientSecure);
     }
 
+    public boolean isZookeeperClientWithEnsembleTracker() {
+        final String defaultValue = String.valueOf(DEFAULT_ZOOKEEPER_CLIENT_ENSEMBLE_TRACKER);
+        final String withEnsembleTracker = getProperty(ZOOKEEPER_CLIENT_ENSEMBLE_TRACKER, defaultValue).trim();
+
+        if (!"true".equalsIgnoreCase(withEnsembleTracker) && !"false".equalsIgnoreCase(withEnsembleTracker)) {
```

Comment:
_Boolean.valueOf()_ can be used instead of comparing against 'true' and 'false' literal string values.

Reply:
Below is the definition of Boolean.valueOf() method. It returns `true` for the string "true" and `false` for any other string. I think it would be better to keep the validation rule strict so that only "true" or "false" is acceptable.
```
    public static Boolean valueOf(String s) {
        return parseBoolean(s) ? TRUE : FALSE;
    }

    public static boolean parseBoolean(String s) {
        return ((s != null) && s.equalsIgnoreCase("true"));
    }
```

Reply:
It looks like other `NiFiProperties` methods have different approaches for `Boolean` handling. The method for `nifi.zookeeper.client.secure` uses this approach, so although it is more strict, it seems acceptable under the circumstances.